source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -1626,7 +1629,7 @@ void COpenGLDriver::drawMeshBuffer(const IGPUMeshBuffer* mb)
     if (!found->nextState.pipeline.graphics.pipeline)
         return;
 
-    found->updateNextState_vertexInput(mb->getVertexBufferBindings(), mb->getIndexBufferBinding().buffer.get(), nullptr, nullptr);
+    found->updateNextState_vertexInput(mb->getVertexBufferBindings(), mb->getIndexBufferBinding().buffer.get(), found->nextState.vertexInputParams.indirectDrawBuf.get(), found->nextState.vertexInputParams.parameterBuf.get());
```

Comment:
@Crisspl why the sudden change here?

When we draw a meshbuffer we shall draw from non-indirect sources, when we want to draw indirect then we'll never use the meshbuffer concept.